api/schemas/User.yaml

```diff
@@ -0,0 +1,56 @@
+# TODO find a way to have these regexes be the source of truth across every component
+type: object
+description: Data requested to login.
+required:
+  - id
+
+properties:
+  id:
+    $ref: "./genericId.yaml"
+
+  mail:
+    type: string
+    format: mail
+    # XXX we do NOT offer the possibility to change the email
+    pattern: ^([a-zA-Z0-9_.+-])+@(([a-zA-Z0-9-])+.)+([a-zA-Z0-9])+$
+    example: mariorossi@gmail.com
+
+  password:
+    type: string
+    format: password
+    # Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character
+    pattern: ^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,32}$
+    minLength: 8
+    maxLength: 32
+    example: ArelativelysecurePassw0rd!
+
+  userData:
+    type: object
+    description: User's personal data
+    required:
+      - firstName
+      - lastName
+      - birthDate
+
+    properties:
+      firstName:
+        type: string
+        minLength: 3
+        example: Mario
+      lastName:
+        type: string
+        example: Rossi
+      birthDate:
+        type: string
+        format: date
+        example: 1990-01-01
+      # these properties should be generated by the client
+      # @createdDate and @lastModifiedDate are set when a row is inserted and/or updated
+      createdDate:
+        type: string
+        format: date-time
+        description: This property is generated automatically by the client.
+      lastModifiedDate:
+        type: string
+        format: date-time
+        description: This property is generated automatically by the client.
```

Comment:
Perchè userData non è uno schema a parte?

Reply:
ho cambiato leggermente la logica con cui separiamo i file, che adesso è basata sulla duplicazione. nel senso, l'unico che utilizza userdata è user quindi è giusto che siano fortemente accoppiati. alla fine queste non sono classi ma schemi di risorse. 

Inoltre nei path che hanno bisogno solo di userdata, basta chiamare $ref: "./schemas/User.yaml#/properties/userData" e il risultato finale è lo stesso ma rimane esplicita l'informazione gerarchica

Reply:
altro bonus: così si bypassa la necessità di mettere la proprietà "id" in userdata perché ogni userdata viene wrappato da un user con solo la proprietà id